test/integration/master/synthetic_master_test.go

```diff
@@ -224,6 +225,120 @@ func TestStatus(t *testing.T) {
 		}
 	}
 }
+func constructBody(val string, size int, field string) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(`{
+		"apiVersion": "apps/v1beta1",
+		"kind": "Deployment",
+		"metadata": {
+		   "name": "test-deployment2",
+		   "namespace": "default",
+		   `)
+
+	switch field {
+	case "labels":
```

Comment:
Compare https://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-go for enums in Golang (that syntax with `iota`). That's a bit cleaner than free form strings.

Reply:
Maybe even more elegant: what do you think about creating a Golang Deployment struct object here and then `json.Marshal` it into json? I imagine the code will be nicer, maybe. Maybe worth a try?

Reply:
@sttts Completely agree with you, will update it.